code/datums/abilities/wizard/summon_staff.dm

```diff
@@ -90,3 +90,92 @@
 				S3.send_staff_to_target_mob(M)
 
 		return 0
+
+/datum/targetable/spell/summon_thunder_staff
+	name = "Summon and Recharge Staff of Thunder"
+	desc = "Returns the staff to your active hand and restores its charges."
+	icon_state = "staff_thunder"
+	targeted = 0
+	cooldown = 20 SECONDS
+	requires_robes = 1
+
+	cast(mob/target)
+		if (!holder)
+			return 1
+
+		var/mob/living/M = holder.owner
+
+		if (!M)
+			return 1
+
+		if (M.getStatusDuration("stunned") > 0 || M.getStatusDuration("weakened") || M.getStatusDuration("paralysis") > 0 || !isalive(M) || M.restrained())
+			boutput(M, "<span class='alert'Not when you're incapacitated or restrained.")
+			return 1
```

Comment:
no need to check for nulls step by step, we can just do it all at once. Also that ugly long if statement didn't stop pins
```suggestion
		var/mob/living/M = holder?.owner
		if (!ismob(M)) 
			return 1
		if (!can_act(M))
			boutput(M, "<span class='alert'Not when you're incapacitated or restrained.")
			return 1
```